### ISSUE TITLE
ThreadPoolScheduler is underdocumented

### ISSUE THREAD

**Issue by Kaiepi** [perl6/doc#2367]
## The problem
At the moment, the only documentation on `ThreadPoolScheduler` is its `new` method. It has many more methods than that.

## Suggestions
Read through `src/core/ThreadPoolScheduler.pm6` and document the missing methods and how to use them.

**Comment by jonathanstowe** [comment]
Though https://docs.perl6.org/type/Scheduler does document partially the interface thar a scheduler should implement - though it might want to document ```loads``` - admittedly this is only useful for implementing a new scheduler.

**Comment by jnthn** [comment]
Yes, `cue` and `uncaught_handler` are the right things to document, and they seem to be already. I note the docs say "methods provided by role Scheduler", which is sort of true I guess, though in reality it's a stubbed method there and actually implemented by concrete implementations of scheduler.

The other thing worth documenting with regards to `ThreadPoolScheduler` is its named arguments `initial_threads` and `max_threads`, which are the number of threads that the scheduler should immediately created and the maximum that ever will be; the scheduler *prefers* to create up to the number of CPU cores and not go beyond that, but it will if it thinks it needs to in order to break a deadlock or in programs that do lots of blocking of threads.

I think the `Awaiter` and `Awaitable` roles will eventually move spec-wards, and then can be documented (but I think they'll be in the 6.e spec rather than the 6.d one; I'd like to see how use of them could work out outside of core before committing to the interface). `ThreadPoolAwaiter` is more in implementation detail territory indeed.

**Comment by Kaiepi** [comment]
The `:hint-affinity` and `:hint-time-sensitive` arguments to `ThreadPoolScheduler.queue` should probably be documented as well

**Comment by JJ** [comment]
To this I would add that the base class is underdocumented: #3321
